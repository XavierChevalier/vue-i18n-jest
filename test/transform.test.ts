import * as fs from 'fs'
import * as path from 'path'
import transformer from '../src/transform'

import { Config } from '@jest/types'

const files = ['default.vue', 'json.vue', 'yaml.vue', 'json5.vue']
const targets = files.map(file => ({
  content: fs.readFileSync(path.resolve(__dirname, `./fixtures/${file}`)).toString(),
  file
}))

const mockConfig = {
  moduleFileExtensions: []
}

test('transform', () => {
  targets.forEach(target => {
    const ret = transformer.process(target.content, target.file, (mockConfig as unknown) as Config.ProjectConfig)
    const code = (typeof ret === 'string' ? ret : ret.code)
    expect(code).toMatchSnapshot(target.file)
  })
})
