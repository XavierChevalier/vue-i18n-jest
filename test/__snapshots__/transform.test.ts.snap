// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform: default.vue 1`] = `
";(function(){
//
//
//
//

export default {
}

})()
var defaultExport = (module.exports.__esModule) ? module.exports.default : module.exports;var __vue__options__ = (typeof defaultExport === \\"function\\"? defaultExport.options: defaultExport)
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"hello!\\")])}
__vue__options__.staticRenderFns = []

//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImRlZmF1bHQudnVlIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBIiwic291cmNlc0NvbnRlbnQiOlsiPHRlbXBsYXRlPlxuICA8cD5oZWxsbyE8L3A+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xufVxuPC9zY3JpcHQ+XG5cbjxpMThuPlxue1xuICBcImphXCI6IHtcbiAgICBcImhlbGxvXCI6IFwi44GT44KT44Gr44Gh44Gv77yBXCJcbiAgfSxcbiAgXCJlblwiOiB7XG4gICAgXCJoZWxsb1wiOiBcImhlbGxvIVwiXG4gIH1cbn1cbjwvaTE4bj4iXX0=;
__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`transform: json.vue 1`] = `
";(function(){
//
//
//
//

export default {
}

})()
var defaultExport = (module.exports.__esModule) ? module.exports.default : module.exports;var __vue__options__ = (typeof defaultExport === \\"function\\"? defaultExport.options: defaultExport)
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"hello!\\")])}
__vue__options__.staticRenderFns = []

//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImpzb24udnVlIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBIiwic291cmNlc0NvbnRlbnQiOlsiPHRlbXBsYXRlPlxuICA8cD5oZWxsbyE8L3A+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xufVxuPC9zY3JpcHQ+XG5cbjxpMThuIGxhbmc9XCJqc29uXCI+XG57XG4gIFwiamFcIjoge1xuICAgIFwiaGVsbG9cIjogXCLjgZPjgpPjgavjgaHjga/vvIFcIlxuICB9LFxuICBcImVuXCI6IHtcbiAgICBcImhlbGxvXCI6IFwiaGVsbG8hXCJcbiAgfVxufVxuPC9pMThuPlxuIl19;
__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`transform: json5.vue 1`] = `
";(function(){
//
//
//
//

export default {
}

})()
var defaultExport = (module.exports.__esModule) ? module.exports.default : module.exports;var __vue__options__ = (typeof defaultExport === \\"function\\"? defaultExport.options: defaultExport)
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"hello!\\")])}
__vue__options__.staticRenderFns = []

//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImpzb241LnZ1ZSJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiO0FBQUE7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQSIsInNvdXJjZXNDb250ZW50IjpbIjx0ZW1wbGF0ZT5cbiAgPHA+aGVsbG8hPC9wPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdD5cbmV4cG9ydCBkZWZhdWx0IHtcbn1cbjwvc2NyaXB0PlxuXG48aTE4biBsYW5nPVwianNvbjVcIj5cbntcbiAgamE6IHtcbiAgICBoZWxsbzogXCLjgZPjgpPjgavjgaHjga/vvIFcIlxuICB9LFxuICBcImVuXCI6IHtcbiAgICBcImhlbGxvXCI6IFwiaGVsbG8hXCJcbiAgfVxufVxuPC9pMThuPlxuIl19;
__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;

exports[`transform: yaml.vue 1`] = `
";(function(){
//
//
//
//

export default {
}

})()
var defaultExport = (module.exports.__esModule) ? module.exports.default : module.exports;var __vue__options__ = (typeof defaultExport === \\"function\\"? defaultExport.options: defaultExport)
__vue__options__.render = function render () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('p',[_vm._v(\\"hello!\\")])}
__vue__options__.staticRenderFns = []

//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbInlhbWwudnVlIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTtBQUNBO0FBQ0E7QUFDQTtBQUNBO0FBQ0E7QUFDQTtBQUNBIiwic291cmNlc0NvbnRlbnQiOlsiPHRlbXBsYXRlPlxuICA8cD5oZWxsbyE8L3A+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0PlxuZXhwb3J0IGRlZmF1bHQge1xufVxuPC9zY3JpcHQ+XG5cbjxpMThuIGxhbmc9XCJ5YW1sXCI+XG5qYTpcbiAgaGVsbG86IOOBk+OCk+OBq+OBoeOBr++8gVxuZW46XG4gIGhlbGxvOiBoZWxsbyFcbjwvaTE4bj4iXX0=;
__vue__options__.__i18n = []
__vue__options__.__i18n.push('{\\"ja\\":{\\"hello\\":\\"こんにちは！\\"},\\"en\\":{\\"hello\\":\\"hello!\\"}}')"
`;
